vector-im/element-android
# #4533:  improve headers UI in room/messages lists

## Type of change


## Content


Improvement for UI of headers sections in the rooms and messages lists. A counter of the list items has been added and the notification badge has been moved to the right.

## Motivation and context



Closes #4533 

## Screenshots / GIFs



New header style : 
<img src="https://user-images.githubusercontent.com/46314705/153207023-fb988975-4e41-464e-860e-d01b39fb3c7c.png" width="300"><img src="https://user-images.githubusercontent.com/46314705/153207032-cef7851e-0a19-4bad-883c-f8397c66e8e9.png" width="300">

## Tests



- Go to the home screen
- Select the rooms tab
- Check headers are correctly displayed with the correct number of items
- Check headers are collapsable/expandable
- Check notification counter is correctly displayed
- Select the messages tab
- Check headers are correctly displayed with the correct number of items
- Check headers are collapsable/expandable
- Check notification counter is correctly displayed
- Open the drawer menu
- Switch space
- Check the lists are correctly updated
- Go to search screen
- Check number of items is correct when changing text search 

## Tested devices

- OS version(s): Android 11

## Checklist
